lib/bill_forward/entities/account.rb

```diff
@@ -7,6 +7,22 @@ module BillForward
   class Account < MutableEntity
   	@resource_path = BillForward::ResourcePath.new("accounts", "account")
 
+    class << self
+      def credit(id, query_object = {}, custom_client = nil)
+        raise ArgumentError.new("id cannot be nil") if id.nil?
+
+        endpoint = sprintf('%s/credit',
+                           ERB::Util.url_encode(id)
+        )
+
+        request_entity = BillForward::GenericEntity.new(
+            query_object
+        )
+
+        self.request_first('post', endpoint, request_entity, nil, custom_client)
```

Comment:
`self` refers to `Account.class`.

`self.request_first` will return `Account.class.request_first`.
This becomes equivalent to invoking:

```ruby
self.request_first_heterotyped(BillForward::Account, 'post', endpoint, request_entity, nil, custom_client)
```